src/services/FileGroupingService.ts

```diff
@@ -1,158 +1,127 @@
+      let file = row.__note__.getFile();
+      const filePath = `${folderPath}/${file.name}`;
+      const fileIsAlreadyInCorrectFolder = row.__note__.filepath === filePath;
+      if (fileIsAlreadyInCorrectFolder) return false;
+      try {
+        await FileGroupingService.createFolder(folderPath);
+      } catch (error) {
+        LOGGER.error(` moveFile Error: ${error.message} `);
+        throw error;
       }
+
+      await app.fileManager.renameFile(file, filePath);
+      row[MetadataColumns.FILE] = `${file.basename}|${folderPath}/${file.name}`;
+      row.__note__.filepath = `${folderPath}/${file.name}`;
+      return true;
+    });
+
+  static createFolder = async (folderPath: string): Promise<void> =>
+    limitCreatingFolders(async () => {
+      await app.vault.adapter.exists(folderPath).then(async (exists) => {
+        if (!exists) {
+          await app.vault.createFolder(`${folderPath}/`);
+        }
+      });
     });
-  }
 
   static organizeNotesIntoSubfolders = async (
     folderPath: string,
     rows: Array<RowDataType>,
     ddbbConfig: LocalSettings,
   ): Promise<number> =>
-    limitBatchDeletionAndOrganization(() =>
-      FileGroupingService._organizeNotesIntoSubfolders(
-        folderPath,
-        rows,
-        ddbbConfig,
-      ),
-    );
-  private static async _organizeNotesIntoSubfolders(
-    folderPath: string,
-    rows: Array<RowDataType>,
-    ddbbConfig: LocalSettings,
-  ): Promise<number> {
-    try {
+    limitBatchDeletionAndOrganization(async () => {
       if (!ddbbConfig.automatically_group_files) return 0;
-      let numberOfMovedFiles = 0;
-      const pathColumns: string[] = (ddbbConfig.group_folder_column || '')
+      const movedRows = [];
+      const pathColumns: string[] = (ddbbConfig.group_folder_column || "")
         .split(",")
         .filter(Boolean);
       for (const row of rows) {
-        let rowTFile = row.__note__.getFile();
-
         const newFilePath = resolveNewFilePath({
           pathColumns,
           row,
           ddbbConfig,
-          folderPath,
+          folderPath
         });
-        // Check if file is already in the correct folder
-        const auxPath = `${newFilePath}/${rowTFile.name}`;
-        const fileIsAlreadyInCorrectFolder = row.__note__.filepath === auxPath;
-        if (fileIsAlreadyInCorrectFolder) continue;
 
-        await FileGroupingService.moveFile(newFilePath, rowTFile);
-        await postMoveFile({ file: rowTFile, row, newFilePath });
-        numberOfMovedFiles++;
+        try {
+          const fileWasMoved = await FileGroupingService.moveFile(newFilePath, row);
+          if (fileWasMoved) {
+            movedRows.push(row);
+          }
+        } catch (error) {
+          new Notice(`Error while moving files into subfolders: ${error.message}`, 5000);
+          throw error;
+        }
       }
-      if (numberOfMovedFiles > 0)
+      if (movedRows.length > 0) {   
+        __set__.current(() => {
```

Comment:
We need to find another way to do this. Global __set__ its a really bad zustand practice. Could provoke shadow re renders

Reply:
Why do you need the set here, to  re render some rows?

Reply:
we need to call another action inside de component to update those urls then. 
On the other hand if you return as argument [...rows] ALL the rows will be rerended instead of just the necessary ones